graphein/datasets/pdb_data.py

```diff
@@ -0,0 +1,1092 @@
+        :rtype: Dict[str, pd.DataFrame]
+        """
+        split_ratios_provided = self.split_ratios is not None
+        assert split_ratios_provided
+
+        # Split clusters
+        log.info(
+            f"Randomly splitting clusters into ratios: {' '.join([str(r) for r in self.split_ratios])}..."
+        )
+        df_splits = self.split_df_proportionally(
+            df,
+            self.splits,
+            self.split_ratios,
+            self.assign_leftover_rows_to_split_n,
+        )
+        log.info("Done splitting clusters")
+
+        # Update splits
+        for split in self.splits:
+            if update:
+                df_split = df_splits[split]
+                if self.df_splits[split] is not None:
+                    self.df_splits[split] = self.merge_df_splits(
+                        self.df_splits[split], df_split
+                    )
+                else:
+                    self.df_splits[split] = df_split
+                df_splits[split] = self.df_splits[split]
+
+        return df_splits
+
+    def cluster(
+        self,
+        min_seq_id: float = 0.3,
+        coverage: float = 0.8,
+        update: bool = False,
+    ) -> Union[pd.DataFrame, Dict[str, pd.DataFrame]]:
+        """Cluster sequences in selection using MMseqs2.
+
+        :param min_seq_id: Sequence identity, defaults to ``0.3``.
+        :type min_seq_id: float, optional
+        :param coverage: Clustering coverage, defaults to ``0.8``.
+        :type coverage: float, optional
+        :param update: Whether to update the selection to the representative
+            sequences, defaults to ``False``.
+        :type update: bool, optional
+        :return: Either a single DataFrame of representative sequences or a
+            Dictionary of split names mapping to DataFrames of randomly-split
+            representative sequences.
+        :rtype: Union[pd.DataFrame, Dict[str, pd.DataFrame]]
+        """
+        # Write fasta
+        self.to_fasta("pdb.fasta")
+        if not is_tool("mmseqs"):
+            log.error(
+                "MMseqs2 not found. Please install it: conda install -c conda-forge -c bioconda mmseqs2"
+            )
+
+        # Create clusters
+        if not os.path.exists("pdb_cluster_rep_seq.fasta"):
+            cmd = f"mmseqs easy-cluster pdb.fasta pdb_cluster tmp --min-seq-id {min_seq_id} -c {coverage} --cov-mode 1"
+            log.info(f"Clustering with: {cmd}")
+            subprocess.run(cmd.split())
+            log.info("Done with clustering")
+
+        # Read fasta
+        df = self.from_fasta(ids="chain", filename="pdb_cluster_rep_seq.fasta")
+        if update:
+            self.df = df
+
+        # Split fasta
+        if self.splits_provided:
+            return self.split_clusters(df, update)
+
+        return df
+
+    def split_df_into_time_frames(
+        self,
+        df: pd.DataFrame,
+        splits: List[str],
+        split_time_frames: List[np.datetime64],
```

Comment:
I have thus far managed to work very little with dates in Python. For my own education, does this work with str formats of `"2020-01-01"` like pandas?